### PR TITLE
updating iCub3 with new arm covers and corresponding options

### PR DESCRIPTION
**Pull request by AlexAntn**
This PR sees the update of the iCub3 with its new upperarm covers.

This change now accounts for skin width.

Other small changes in the model options were introduced (collision/contact sensors for the upper arms; small tweak on upper joint limit for the roll to be on par with real robot).

### CODE REVIEW

**Review by Nicogene — APPROVED** [review]
LGTM, we can merge it if the generated model has been tested

**Comment by AlexAntn** [comment]
I have tested the model, it is working properly in Gazebo

File: simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv
Comment (traversaro) [inline simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: Did we documented somewhere this limits? It would be great if we could have some official documentation, such as https://icub-tech-iit.github.io/documentation/icub_kinematics/icub-joints-limits/icub-joints-limits/ , so that here we just reference those.

File: simmechanics/data/icub3/ICUB_3_all_options.yaml
Comment (traversaro) [inline simmechanics/data/icub3/ICUB_3_all_options.yaml:756]: Are you sure that it is necessary to duplicate this infor from the model manually? If you see how we inject collision info for `r_foot_rear`, this does not seems to be necessary, as  in the `collision` tag only the elements that are added need to be included. This avoids the need for manually updating the `origin` tag if the transform location changes.

**Comment by traversaro** [comment]
Friendly ping @AlexAntn .

File: simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv
Comment (AlexAntn) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: I don't think we did yet, since there was still going to be an update of the limits for the real robot, but we can update them I think

File: simmechanics/data/icub3/ICUB_3_all_options.yaml
Comment (AlexAntn) [reply simmechanics/data/icub3/ICUB_3_all_options.yaml]: Indeed we don't need the duplicated section, I have removed it, tested it and it works properly, I have committed and updated the PR (waiting for tests)

**Comment by traversaro** [comment]
@AlexAntn is this PR ready to be merged? For the the future, I don't get notifications for new commits, so feel free to  comment whenever the PR is ready for review/merge.

File: simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv
Comment (traversaro) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: If they are not updated, I am ok in merging this PR before the documentation changed, but I would open an issue on the update of joint limits, otherwise we may forget about it.

File: simmechanics/data/icub3/ICUB_3_all_options_gazebo.yaml.in
Comment (traversaro) [inline simmechanics/data/icub3/ICUB_3_all_options_gazebo.yaml.in:839]: The duplication problem seems still here?

**Review by traversaro — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by pattacini** [comment]
Hi @AlexAntn 

We are going to regenerate the URDF of iCub 3 quite soon.
It'd be great if we could include also these new additions to the model.

File: simmechanics/data/icub3/ICUB_3_all_options_gazebo.yaml.in
Comment (AlexAntn) [reply simmechanics/data/icub3/ICUB_3_all_options_gazebo.yaml.in]: I didn't check the gazebo options - I have now removed them and committed the changes!

File: simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv
Comment (AlexAntn) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: Regarding this point I think we need to discuss with @pattacini about these limits. Were they set for iCub3 anywhere, since the joint calibration? Where should we get these values from?

Comment (pattacini) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: @traversaro's suggestions were:
1. to update our official documentation website with the iCub 3 joint limits.
2. to create in robots-configuration an issue in order to keep track of the need for checking these new limits.

I think you can go ahead certainly with 2; it'd be great to have also 1 done.

Comment (AlexAntn) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: I have opened a new issue in robots-configuration, that we can see here: https://github.com/robotology/robots-configuration/issues/280

this should allow us to keep track of this, in order to update them in the future

Comment (traversaro) [reply simmechanics/data/icub3/ICUB_3_joint_all_parameters.csv:6]: Thanks!

**Comment by pattacini** [comment]
@traversaro do you deem we can merge this PR or are there still open points?

**Comment by traversaro** [comment]
> 
> 
> @traversaro do you deem we can merge this PR or are there still open points?

My comments were solved, but from https://github.com/robotology/icub-models-generator/pull/202#issuecomment-877586812 it seemed that you wanted to wait before merging this. If this is not the case anymore, we can merge.

**Review by traversaro — APPROVED** [review]

**Comment by pattacini** [comment]
>My comments were solved, but from #202 (comment) it seemed that you wanted to wait before merging this. If this is not the case anymore, we can merge.

We can merge the PR so that we will inherit these additions into the URDF next time.

**Pull request merged**
